python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -1549,55 +1647,64 @@ def _get_yml_file_data(self, config_descriptor):
-            # so we will stat all of the top level env files that we find. Note
-            # that this means we are not covering included yml files, but we
-            # accept that for the sake of speed.
             yml_files = dict()
 
             if root_path is not None:
-                config_path = os.path.join(root_path, "config", "env")
-
-                # We have a case where the descriptor API has changed during the
-                # development of this v2 RPC API in terms of what the root path
-                # is that's returned from the config descriptor's get_path method.
-                # At one point, we got the full path to the "config" directory, and
-                # later on it was switched to path to one directory above that, at
-                # the root of the config (where the tank command is). We can pretty
-                # easily check both, and we'll go with the more likely case, which
-                # is the newer of the two conventions, before checking the other.
-                #
-                # It's worth noting that it was changed because we considered the
-                # previous behavior to be incorrect, and unintentional. Rooting at
-                # the config root (where the tank command resides) is the more
-                # correct behavior.
-                if not os.path.exists(config_path):
-                    config_path = os.path.join(root_path, "env")
+                config_path = self._get_config_env_root(root_path)
 
                 logger.debug(
                     "Config %s is mutable -- environment file mtimes will be used to determine cache validity.",
                     config_path,
                 )
 
-                for file_path in glob.glob(os.path.join(config_path, "*.yml")):
-                    logger.debug("Checking mtime: %s", file_path)
-                    yml_files[file_path] = os.path.getmtime(file_path)
+                # We do a deep scan of from the config's "env" root down to
+                # its bottom.
+                for root, dir_names, file_names in os.walk(config_path):
+                    for file_name in fnmatch.filter(file_names, "*.yml"):
+                        full_path = os.path.join(root, file_name)
+                        yml_files[full_path] = os.path.getmtime(full_path)
 
             logger.debug(
-                "Contents hash computed using %s yml files -- shallow scan, no subdirectories.",
+                "Contents hash computed using %s yml files.",
                 len(yml_files),
             )
 
             logger.debug("Files checked for mtime: %s", sorted(yml_files.keys()))
-
             self._cache.setdefault(self.YML_FILE_DATA, dict())[root_path] = yml_files
         else:
             logger.debug("Cached yml file data found for %r.", config_descriptor)
 
         return self._cache[self.YML_FILE_DATA].get(root_path)
 
+    def _get_config_env_root(self, config_root_path):
```

Comment:
This is just making some common logic shared between the two yml file getters above reusable.